src/Raven.Server/Routing/RouteInformation.cs

```diff
@@ -96,14 +121,23 @@ public Task CreateDatabase(RequestHandlerContext context)
                 if (context.Database == null)
                     DatabaseDoesNotExistException.Throw(databaseName.Value);
 
-                return context.Database?.DatabaseShutdown.IsCancellationRequested == false
-                    ? Task.CompletedTask
-                    : UnlikelyWaitForDatabaseToUnload(context, context.Database, databasesLandlord, databaseName);
+                if (context.Database?.DatabaseShutdown.IsCancellationRequested == false)
+                {
+                    ClearIdleDatabase(context, context.Database.Name);
+                    return Task.CompletedTask;
+                }
+
+                return UnlikelyWaitForDatabaseToUnload(context, context.Database, databasesLandlord, databaseName);
             }
 
             return UnlikelyWaitForDatabaseToLoad(context, database, databasesLandlord, databaseName);
         }
 
+        private void ClearIdleDatabase(RequestHandlerContext context, string databaseName)
+        {
+            context.RavenServer.ServerStore.IdleDatabases.Remove(databaseName, out _);
```

Comment:
There is no `Remove` method in ConcurrentDictionary. I'm afraid that this might be some extension method? Use `TryRemove` instead.

Reply:
done